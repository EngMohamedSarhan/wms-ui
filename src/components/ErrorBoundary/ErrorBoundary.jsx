import PropTypes from "prop-types";
import React from "react";
import { connect } from "react-redux";

import { openNotification } from "../../redux/reducers/settingsSlice";
import AlertModal from "../AlertModal/AlertModal";

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
      errorInfo: "",
    };
  }

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  render() {
    return this.state.hasError ? (
      <AlertModal
        data-testid="error-boundary"
        isOpen
        title="ERROR"
        actionLabel="Refresh"
        onAction={() => window.location.reload()}
      >
        An error has occured please refresh the page!
      </AlertModal>
    ) : (
      this.props.children
    );
  }
}

// const mapDispatchToProps = (dispatch) => ({ openNotification });

export default connect(null, { openNotification })(ErrorBoundary);

ErrorBoundary.propTypes = {
  children: PropTypes.node,
  openNotification: PropTypes.func,
};
