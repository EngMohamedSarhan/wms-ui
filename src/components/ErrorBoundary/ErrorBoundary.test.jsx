/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import React from "react";

import ErrorBoundary from "./ErrorBoundary";

describe("ErrorBoundary Component", () => {
  test("ErrorBoundary rendering", () => {
    const ThrowError = () => {
      throw new Error("Error!");
    };
    const { getByTestId } = render(
      <ErrorBoundary>
        <ThrowError />
      </ErrorBoundary>
    );

    expect(getByTestId("error-boundary")).toBeInTheDocument();
  });
});

describe("ErrorBoundary Component", () =>
  test("ErrorBoundary snapshot", () => {
    const renderer = new ShallowRenderer();
    const ThrowError = () => {
      throw new Error("Error!");
    };
    renderer.render(
      <ErrorBoundary>
        <ThrowError />
      </ErrorBoundary>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  }));
