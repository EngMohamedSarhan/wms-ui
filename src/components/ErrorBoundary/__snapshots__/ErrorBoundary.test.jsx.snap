// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component ErrorBoundary snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Memo(Connect(ErrorBoundary))>
    <ThrowError />
  </Memo(Connect(ErrorBoundary))>
</Context.Provider>
`;
